docs/content/Schema/Fundamentals/Concepts.md

```diff
@@ -0,0 +1,303 @@
+      // This is required for de-duplicating results when using joins
+      primaryKey: true
+    },
+    status: {
+      sql: `status`,
+      type: `string`
+    },
+  },
+});
+```
+
+[The `line_items` table](#top) also has a couple of dimensions which can be
+represented as follows:
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  dimensions: {
+    id: {
+      sql: `id`,
+      type: `number`,
+      // Again, we explicitly let Cube.js know this field is the primary key
+      // This is required for de-duplicating results when using joins
+      primaryKey: true
+    },
+  },
+});
+```
+
+Dimensions can be of different types, and you can find them all
+[here][ref-schema-dimension-types].
+
+### Time Dimensions
+
+Time-based properties should be represented as dimensions with type `time`. Time
+dimensions allow granulating the result set for queries sent to Cube.js.
+
+We can add the necessary time dimensions to both schemas as follows:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+
+    completedAt: {
+      sql: `completed_at`,
+      type: `time`,
+    },
+  },
+});
+```
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  dimensions: {
+
+    ...,
+
+    createdAt: {
+      sql: `created_at`,
+      type: `time`
+    },
+  },
+});
+```
+
+Time dimensions are essential to enabling performance boosts such as
+[partitioned pre-aggregations][ref-caching-use-preaggs-partition-time] and
+[incremental refreshes][ref-tutorial-incremental-preagg].
+
+## Measures
+
+Measures represent the properties of a **set of data points** in the cube. To
+add a measure called `count` to our `Orders` cube, for example, we can do the
+following:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  measures: {
+    count: {
+      type: `count`,
+    },
+  },
+});
+```
+
+In our `LineItems` cube, we can We could also create a measure to sum up the
+total value of line items sold:
+
+```javascript
+cube('LineItems', {
+
+  ...,
+
+  measures: {
+    total: {
+      sql: `price`,
+      type: `sum`,
+    },
+  },
+})
+```
+
+Measures can be of different types, and you can find them all
+[here][ref-schema-measure-types].
+
+## Joins
+
+Joins define the relationships between cubes, which then allows accessing and
+comparing properties from two or more cubes at the same time. In Cube.js, all
+joins are `LEFT JOIN`s by default.
+
+<!-- prettier-ignore-start -->
+[[info | ]]
+| An `INNER JOIN` can be replicated with Cube.js; when making a Cube.js query,
+| add a filter for `IS NOT NULL` on the required column.
+<!-- prettier-ignore-end -->
+
+In the following example, we are left-joining the `LineItems` cube onto our
+`Orders` cube:
+
+```javascript
+cube('Orders', {
+
+  ...,
+
+  joins: {
+    LineItems: {
+      relationship: `belongsTo`,
+      sql: `${CUBE}.id = ${LineItems}.order_id`,
```

Comment:
Suddenly, we're using `${CUBE}` to reference the cube. I wonder if there should be a place in the docs (maybe not on this page) where we recommend always using `${CUBE}` or a specific cube reference to prevent name clashes.

Reply:
Also, it's great to explain `${CUBE}.column` vs `${CUBE.member}`.

Reply:
Agreed that we should explain the above; but also not sure if this is the best place to do it. Also, do you think I should update _all_ the examples on this page to use the `CUBE` global?